src/ClapDetector.js

```diff
@@ -0,0 +1,168 @@
+                    if (devices[i].kind === 'audioinput') deviceIds.push(devices[i].deviceId);
+                }
+                this.setState({
+                    contextAvailable : true,
+                    audioInputDeviceIds : deviceIds
+                });
+            }, (error) => {
+                console.log(error.name);
+                console.log(error.message);
+            });
+        }
+    }
+
+    componentDidUpdate(prevProps: {}, prevState: ClapDetectorState) {
+        if (this.state.audioInputDeviceIds !== prevState.audioInputDeviceIds) {
+            if (this.state.audioInputDeviceIds.length === 0) {
+                console.log('state not updated');
+            } else {
+                const AudioContext = window.AudioContext || window.webkitAudioContext;
+                const context = new AudioContext();
+
+                Promise.all([
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[0], context),
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[1], context),
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[3], context)
+                ]).then((analysers) => {
+                    const defaultInputAnalyser = analysers[0];
+                    const externalInputAnalyser = analysers[1];
+                    const wirelessInputAnalyser = analysers[2];
+                    const scriptProcessorAnalysisNode = context.createScriptProcessor(2048,2,1);
+                    const audioGain = context.createGain();
+                    scriptProcessorAnalysisNode.connect(audioGain);
+                    audioGain.connect(context.destination);
+                    scriptProcessorAnalysisNode.onaudioprocess = () => {
+
+                        let sim1 = this.getClapSimilarity(defaultInputAnalyser, 5, 1);
+                        let sim2 = this.getClapSimilarity(externalInputAnalyser, 5, 2);
+                        let sim3 = this.getClapSimilarity(wirelessInputAnalyser, 5, 3);
+
+                        if (sim1 >= sim2 && sim1 >= sim3 && sim1 >= 0.90) {
+                            console.log(sim1);
+                            console.log('left');
+                            this.props.onDetect('left');
+                        } else if (sim2 >= sim1 && sim2 >= sim3 && sim2 >= 0.90) {
+                            console.log(sim2);
+                            console.log('right');
+                            this.props.onDetect('right');
+                        } else if (sim3 >= sim1 && sim3 >= sim2 && sim3 >= 0.90) {
+                            console.log(sim3);
+                            console.log('forward');
+                            this.props.onDetect('forward');
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    addAudioStreamNode = async (deviceId: string, audioContext: any): any=> {
```

Comment:
I think `deviceId` isn't used and we always use `this.state.audioInputDeviceIds[0]`